wire-ios/Wire-iOS/Sources/UserInterface/Settings/AccentColorPickerController.swift

```diff
@@ -17,37 +17,30 @@
 //
 
 import SwiftUI
-import UIKit
-import WireCommonComponents
-import WireDataModel
-import WireSyncEngine
+import protocol WireDataModel.EditableUserType
+import protocol WireSyncEngine.UserSession
+import enum WireUtilities.AccentColor
 
-class AccentColorPickerHostingController: UIHostingController<ColorPickerView> {
+final class AccentColorPickerController: UIHostingController<ColorPickerView> {
 
+    @available(*, unavailable)
     required init?(coder aDecoder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
 
-    init(selectedAccentColor: Binding<AccentColor?>) {
-        let allAccentColors = AccentColor.allCases
-        var initialSelectedColor: AccentColor?
+    init(selfUser: EditableUserType, userSession: UserSession) {
+        let accentColor = AccentColor(rawValue: selfUser.accentColorValue) ?? .default
 
-        if let bindingValue = selectedAccentColor.wrappedValue {
-            // If the binding has an initial value, use it
-            initialSelectedColor = bindingValue
-        } else if let firstColor = allAccentColors.first {
-            // If no initial value, use the first available color as a default
-            initialSelectedColor = firstColor
-        }
-
-        super.init(rootView: ColorPickerView(selectedColor: initialSelectedColor, colors: allAccentColors, onColorSelect: { selectedColor in
-            selectedAccentColor.wrappedValue = selectedColor
-            if let colorIndex = allAccentColors.firstIndex(of: selectedColor) {
-                ZMUserSession.shared()?.perform {
-                    ZMUser.selfUser()?.accentColorValue = allAccentColors[colorIndex].rawValue
+        let colorPickerView = ColorPickerView(
+            selectedColor: accentColor,
+            colors: AccentColor.allCases,
+            onColorSelect: { accentColor in
+                userSession.perform {
```

Comment:
What about moving this code out? Then `AccentColorPickerController` could probably be deleted (replaced by `UIHostingController<ColorPickerView>` instead of subclassed from).

Reply:
I've thought about it, where should the logic exist? What is responsible to init the SwiftUI with the right closure and handle the result on the user session?
I am not sure if the creating factory is the right place.

Maybe it would be better to create a `ViewModel`?

Reply:
After talking we decided to keep it like it is for now.